vhost-user-backend/src/backend.rs

```diff
@@ -177,6 +180,9 @@ pub trait VhostUserBackendMut: Send + Sync {
     /// Get available vhost protocol features.
     fn protocol_features(&self) -> VhostUserProtocolFeatures;
 
+    /// Reset the emulated device state.
+    fn reset_device(&mut self) {}
```

Comment:
Might it make sense to return an error if the backend-specific implementation for some reason does not support reset?

Although maybe just managing the feature is enough, at which point the backend doesn't enable it, also because from the specification it doesn't seem to be expected to fail.

Reply:
I think managing the feature is enough. Beyond that, I think reset should be an infallible operation; so if it’s generally supported, it should never fail in such a way where you’d return an error to qemu.